### ISSUE TITLE
Feature Request - Allow configuration of "key-type" or "Signature Algorithm"

### ISSUE THREAD

**Issue by GoTTi74** [hassio-addons/addon-nginx-proxy-manager#610]
# Problem/Motivation

> When using an old web-server (e.g. Synology DSM 6.x) the ECC/ECDSA "Signature Algorithm" is not supported. Instead the former RSA standard is required.

## Expected behavior

> Allow NGINX to configure the default "Signature Algorithm"

## Actual behavior

> By default, the "signature algorithm" is configured in ./etc/letsencrypt.ini as 'key-type = ecdsa'. When issuing a certificate with this key-type, it can't be used in older versions of Synology DSM (6.x). Instead, Synology will show an error "DSM does not support ECC certificates" when trying to import a certificate managed by Nginx. 

## Steps to reproduce

> To workaround this issue, one must manually update the 'key-type = ecdsa' to 'key-type = rsa' before renewing a certificate required by Synology (e.g. for "Synology Drive Sync").

## Proposed changes

> The manual workaround is possible but not persistent.
> Ideally, one could configure the preferred "key-type" in the AddOn

**Comment by averstappen** [comment]
I have a similar issue, in my case a printer that doesnt support ecdsa. My prefered solution would be to be able to override the key-type per certificate, so that all other certificates stay at the default.

**Comment by GoTTi74** [comment]
> I hope and wish, like GoTTI74, that Synology makes a change to allow easy configuration to use the preferred certificate of the program that requires it in a form other than ECDSA or RSA.

You must have misunderstood me. My hope is to be able to configure Nginx to either receive an ECDSA or RSA certificate. I'm not expecting that Synology makes a change.

**Comment by GoTTi74** [comment]
Correct, I am able to manually change the configuration but as soon as the Docker container is restarted, it’s reverted to the default configuration. Hence the wish to be able to configure the type rather than have the current being static. On Jan 13, 2025, at 09:39, mbisax ***@***.***> wrote:﻿
Thanks for your correction.
However, I believe that what you expect, without an intervention from Synology, is not possible. I tried to change the Proxy-Reverse configuration but, as soon as the service is re-run, the configuration returns to the default. Maybe I made some mistake but that's what happens to me.
In any case, you're right, I misunderstood what you meant: it would be great if it were possible to decide which type of certificate to use.
